diff --git a/src/Entity/Access/GroupContentAccessControlHandler.php b/src/Entity/Access/GroupContentAccessControlHandler.php
index fd69e16..cb9fc35 100644
--- a/src/Entity/Access/GroupContentAccessControlHandler.php
+++ b/src/Entity/Access/GroupContentAccessControlHandler.php
@@ -27,7 +27,12 @@ class GroupContentAccessControlHandler extends EntityAccessControlHandler {
    */
   protected function checkCreateAccess(AccountInterface $account, array $context, $entity_bundle = NULL) {
     $group_content_type = GroupContentType::load($entity_bundle);
-    return $group_content_type->getContentPlugin()->createAccess($context['group'], $account);
+    $content_type_plugin = $group_content_type->getContentPlugin();
+
+    if (!empty($context['create_mode']) && $content_type_plugin->definesEntityAccess()) {
+      return $content_type_plugin->createEntityAccess($context['group'], $account);
+    }
+    return $content_type_plugin->createAccess($context['group'], $account);
   }

 }
diff --git a/src/Entity/Controller/GroupContentController.php b/src/Entity/Controller/GroupContentController.php
index ec2192c..ac45868 100644
--- a/src/Entity/Controller/GroupContentController.php
+++ b/src/Entity/Controller/GroupContentController.php
@@ -107,7 +107,7 @@ class GroupContentController extends ControllerBase {
     // Filter out the bundles the user doesn't have access to.
     $access_control_handler = $this->entityTypeManager->getAccessControlHandler('group_content');
     foreach ($bundle_names as $plugin_id => $bundle_name) {
-      $access = $access_control_handler->createAccess($bundle_name, NULL, ['group' => $group], TRUE);
+      $access = $access_control_handler->createAccess($bundle_name, NULL, ['group' => $group, 'create_mode' => $create_mode], TRUE);
       if (!$access->isAllowed()) {
         unset($bundle_names[$plugin_id]);
       }
